src/main/scala/com/cognite/spark/datasource/FilesRelation.scala

```diff
@@ -25,64 +26,34 @@ case class FileItem(
     createdTime: Option[Long],
     lastUpdatedTime: Option[Long])
 
-case class UpdateFileItem(
-    id: Option[Long],
-    directory: Option[Setter[String]],
-    source: Option[Setter[String]],
-    sourceId: Option[Setter[String]],
-    fileType: Option[Setter[String]],
-    metadata: Option[Setter[Map[String, String]]],
-    assetIds: Option[Map[String, Seq[Long]]]
-)
-object UpdateFileItem {
-  def apply(fileItem: FileItem): UpdateFileItem =
-    new UpdateFileItem(
-      fileItem.id,
-      Setter[String](fileItem.directory),
-      Setter[String](fileItem.source),
-      Setter[String](fileItem.sourceId),
-      Setter[String](fileItem.fileType),
-      Setter[Map[String, String]](fileItem.metadata),
-      fileItem.assetIds.map(a => Map("set" -> a))
-    )
-}
-
 class FilesRelation(config: RelationConfig)(val sqlContext: SQLContext)
-    extends CdpRelation[FileItem](config, "files")
+    extends SdkV1Relation[File, Files[IO], FileItem](config, "files")
     with InsertableRelation {
 
-  override val fieldsWithPushdownFilter: Seq[String] =
-    Seq("assetId", "dir", "name", "fileType", "source")
-
-  override def insert(data: DataFrame, overwrite: Boolean): Unit =
-    data.foreachPartition((rows: Iterator[Row]) => {
-      implicit val contextShift: ContextShift[IO] = IO.contextShift(ExecutionContext.global)
-      val batches = rows.grouped(config.batchSize.getOrElse(Constants.DefaultBatchSize)).toVector
-      batches.grouped(Constants.MaxConcurrentRequests).foreach { batchGroup =>
-        batchGroup.parTraverse(postFiles).unsafeRunSync()
-      }
-      ()
-    })
-
-  def postFiles(rows: Seq[Row]): IO[Unit] = {
-    val fileItems = rows.map { r =>
-      val fileItem = fromRow[FileItem](r)
-      fileItem.copy(metadata = filterMetadata(fileItem.metadata))
+  override def getFromRowAndCreate(rows: Seq[Row]): IO[Seq[File]] = {
+    val fileUpdates = rows.map { r =>
+      val file = fromRow[File](r)
+      fileUpdateFromFile(file.copy(metadata = filterMetadata(file.metadata)))
     }
+    client.files.update(fileUpdates)
```

Comment:
Couldn't we use `updateFromRead`, after filtering the metadata?

Reply:
Makes a lot of sense, thanks!